addons/html_builder/static/src/plugins/language_selector_option_plugin.js

```diff
@@ -0,0 +1,113 @@
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+import { withSequence } from "@html_editor/utils/resource";
+import {
+    LanguageSelectorOption,
+    LANGUAGE_SELECTOR_STYLES,
+    LANGUAGE_SELECTOR_LABELS,
+} from "./language_selector_option";
+import { after } from "@html_builder/utils/option_sequence";
+import { rpc } from "@web/core/network/rpc";
+import { HEADER_BORDER } from "@html_builder/website_builder/plugins/options/header_option";
+
+const LANGUAGE_SELECTOR = after(HEADER_BORDER);
+class LanguageSelectorOptionPlugin extends Plugin {
+    static id = "languageSelectorOption";
+    resources = {
+        reload_targets: ["#wrapwrap > header nav.navbar .o_header_language_selector"],
+        builder_options: [
+            withSequence(LANGUAGE_SELECTOR, {
+                OptionComponent: LanguageSelectorOption,
+                editableOnly: false,
+                selector: "#wrapwrap > header nav.navbar .o_header_language_selector",
+                groups: ["website.group_website_designer"],
+            }),
+        ],
+        builder_actions: this.getActions(),
+    };
+
+    setup() {
+        this.styleValue = "";
+        this.labelValue = "";
+        this.styleValues = LANGUAGE_SELECTOR_STYLES.map((style) => style.value);
+        this.labelValues = LANGUAGE_SELECTOR_LABELS.map((label) => label.value);
+        this.styleKeys = this.getDataKeys(this.styleValues);
+        this.labelKeys = this.getDataKeys(this.labelValues);
+    }
+
+    getActions() {
+        return {
+            customizeStyle: {
+                reload: {},
+                prepare: async () => {
+                    this.styleValue = await this.getEnabledCustomizeValues(
+                        this.styleValues,
+                        this.styleKeys
+                    );
+                },
+                apply: async ({ value }) => {
+                    await this.customizeWebsiteData(value, this.styleKeys);
+                },
+                isApplied: ({ value }) => value === this.styleValue,
+            },
+            customizeLabel: {
+                reload: {},
+                prepare: async () => {
+                    this.labelValue = await this.getEnabledCustomizeValues(
+                        this.labelValues,
+                        this.labelKeys
+                    );
+                },
+                apply: async ({ value }) => {
+                    await this.customizeWebsiteData(value, this.labelKeys);
+                },
+                isApplied: ({ value }) => value === this.labelValue,
+            },
+        };
```

Comment:
I believe you can just rely on the `websiteConfig` action like everything else to enable/disable views.